libraries/AdaptiveExpressions/ExpressionFunctions.cs

```diff
@@ -3764,6 +3763,41 @@ private static IEnumerable<object> Flatten(IEnumerable<object> list, int dept)
                     },
                     ReturnType.Number,
                     expr => ValidateArityAndAnyType(expr, 1, 1, ReturnType.String)),
+                new ExpressionEvaluator(
+                    ExpressionType.DateTimeDiff,
+                    (expr, state, options) =>
+                    {
+                        object dateTimeStart = null;
+                        object dateTimeEnd = null;
+                        object value = null;
+                        string error = null;
+                        IReadOnlyList<object> args;
+                        (args, error) = EvaluateChildren(expr, state, options);
+                        if (error == null)
+                        {
+                            if (args[0] is string ts1 && args[1] is string ts2)
+                            {
+                                (dateTimeStart, error) = Ticks(ts1);
+                                if (error == null)
+                                {
+                                    (dateTimeEnd, error) = Ticks(ts2);
+                                }
+                            }
+                            else
+                            {
+                                error = $"{expr} can't evaluate.";
```

Comment:
>error = $"{expr} can't evaluate."; [](start = 32, length = 34)

Better error message would be "{expr} must have two ISO timestamps."

Reply:
error message updated